### PR TITLE
Add favourites api endpoint

### PR DESCRIPTION
**Pull request by humrochagf**
Hey, this adds the endpoint in the API to fetch favourites.

I'm still trying to get the workflow of takahe's model/service/view so if I misplaced something just let me know :slightly_smiling_face:

### CODE REVIEW

**Comment by humrochagf** [comment]
Another thing was that I named it as `favourites` to match the expected mastodon API, but we can rename internal functions the way it works best

**Comment by humrochagf** [comment]
Also tested it with both Megalodon and Tusky apps :slightly_smiling_face: 
megalodon_favs
tusky_favs

File: api/views/timelines.py [hunk 1]
```diff
@@ -144,3 +144,36 @@ def conversations(
...
+        limit=limit,
+    )
+    # Convert those to the JSON form
+    pager.jsonify_posts(identity=request.identity)
+
+    # Add the link header if needed
+    if pager.results:
+        response.headers["Link"] = pager.link_header(
+            request,
+            ["limit", "local", "remote", "only_media"],
```
Comment (andrewgodwin) [inline api/views/timelines.py:176]: This should just be "limit", as that's the only option on the endpoint

File: activities/services/timeline.py [hunk 2]
```diff
@@ -82,3 +82,16 @@ def identity_public(self, identity: Identity):
             .unlisted(include_replies=True)
             .order_by("-id")
         )
+
+    def favourites(self) -> models.QuerySet[Post]:
```
Comment (andrewgodwin) [inline activities/services/timeline.py:86]: Our convention is to call these `likes` everywhere except where we interface with Mastodon - so here and the tests should use like, and the mastodon API is the thing that's called favourites.

**Review by andrewgodwin — COMMENTED** [review]
Looks pretty good, couple of small things!

File: api/views/timelines.py (hunk 1 above)
Comment (humrochagf) [reply api/views/timelines.py]: fixed

File: activities/services/timeline.py (hunk 2 above)
Comment (humrochagf) [reply activities/services/timeline.py]: Alright, all updated :slightly_smiling_face:

**Review by andrewgodwin — APPROVED** [review]

**Comment by andrewgodwin** [comment]
Thanks for this!

**Pull request merged**